v8/third_party/heap/mmtk/mmtk.cc

```diff
@@ -117,7 +117,15 @@ AllocationResult Heap::Allocate(size_t size_in_bytes, AllocationType type, Alloc
   int space = (type == AllocationType::kCode) ? 3 : (type == AllocationType::kReadOnly) ? 4 : (large_object) ? 2 : 0;
   Address result =
       reinterpret_cast<Address>(alloc(tph_mutator_, size_in_bytes, align_bytes, 0, space));
-  tph_archive_insert(tph_data_->archive(), reinterpret_cast<void*>(result), tph_data_->isolate(), uint8_t(space));
+  AllocationSpace allocation_space;
```

Comment:
If I understand correct, the `space` above is actually the `AllocationSemantics` in mmtk, and this `allocation_space` is the space from V8 runtime. Is this correct? The reason why you would need to save the `allocation_space` for each address result is that mmtk-core does not provide a function to check whether an address is in a certain space, is this right?

Reply:
Yes, `space` is MMTk `AllocationSemantics` and `allocation_space` is the v8 space that the runtime expects the allocation to happen.

The reason to save `allocation_space` is that v8 wants to check `allocation_space` later after allocation. This usually happens in unit tests -- a lot of them add explicit checks to assert some object is located in code space or old space. Until we add all of these spaces to MMTk common plan, I have to store them in a table for later lookups.

(some compiler optimizations and heap serialization code also want to check the `allocation_space` for a given object.)

Reply:
OK. I would suggest adding some comments to explain this. Calling allocation semantics as 'space' and assigning it to magic numbers is confusing already, and there is another allocation space.

Reply:
Done